### PR TITLE
Better mode inputs

### PR DESCRIPTION
**Pull request by Geeber**
Changes:

* Removed dart mode
* Added rainbow and (a fairly hacky attempt at) fire patterns to beachball mode
* Fixed the strobe mode speed controls to (a) be continuous and (b) follow intuition vis-a-vis up/down
* Improve inputs of flicker mode
* Make barrel-specific rendering of rainbow mode (this actually makes the D-pad inputs fairly interesting to play with)

I unfortunately didn't get around to adding wipes to rainbow mode, but I think this leaves things in a pretty good state for the burn.

### CODE REVIEW

File: config.clj [hunk 1]
```diff
@@ -96,9 +96,14 @@
   [color/white color/red color/white color/yellow color/white color/blue])
 (def american-colors
   [color/blue color/blue color/blue color/white color/red color/white color/red color/white color/red color/white])
+(def rainbow-colors
+  (map color/rainbow (range 0 1 0.1)))
+(def fire-colors
+  (let [colors [(color/rgb 156/256 42/256 34/256) (color/rgb 226/256 87/256 34/256) (color/rgb 1.0 73/256 73/256) (color/rgb 1.0 189/256 91/256) (color/rgb 253/256 207/256 88/256)]]
+    (repeatedly 12 #(rand-nth colors))))
```
Comment (greglook) [inline config.clj]: Style nit: usually if you're defining something in a `let` block that's only used in one place, that should be a code smell.

Comment (greglook) [inline config.clj]: Also, was it your intent that this color sequence is statically "randomized" on each system boot?

File: src/clojure/playasophy/wonderdome/mode/flicker.clj [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 
-(def ^:const ^:private brightness-rate 0.027)
+(def ^:const ^:private brightness-rate 0.05)
 
 ; TODO: 'speed' is more like 'delay'
-(def ^:const ^:private min-speed 10)
+(def ^:const ^:private min-speed 5)
 (def ^:const ^:private max-speed 3500)
-(def ^:const ^:private speed-rate 100)
+(def ^:const ^:private speed-rate -200)
```
Comment (greglook) [inline src/clojure/playasophy/wonderdome/mode/flicker.clj]: Negative to invert the axis direction?

**Comment by greglook** [comment]
Two questions, otherwise looks good! Will test out on the hardware.

File: config.clj (hunk 1 above)
Comment (Geeber) [inline config.clj]: OK, fair enough on the style nit. This was a refactoring of a refactoring of an initial attempt that was *much* more stylistically objectionable ;) So would something like this be more idiomatic?

```clojure
(def fire-colors
  (repeatedly 12 #(rand-nth
                    [(color/rgb 156/256 42/256 34/256)
                     (color/rgb 226/256 87/256 34/256)
                     (color/rgb 1.0 73/256 73/256)
                     (color/rgb 1.0 189/256 91/256)
                     (color/rgb 253/256 207/256 88/256)]))))
```

And yes, the randomization on each boot was my (ridiculously simplistic) way to add "entropy" to this mode. This was a really spur-of-the-moment thought I had as I was wrapping up last night: what is the most ridiculously simple way I can try to make a vaguely fire-like effect with one of the existing modes, in 10 minutes' coding or less? If you'd rather just make it totally static, or remove it entirely, that's cool. I also have no sense of how well/badly this effect will work on the actual lantern, so might be worth looking at that and then making a call.

File: src/clojure/playasophy/wonderdome/mode/flicker.clj (hunk 2 above)
Comment (Geeber) [inline src/clojure/playasophy/wonderdome/mode/flicker.clj]: Yeah. Probably better to refactor the variable to mean the inverse of what it does currently, but that was more work. This was quick and dirty.

**Pull request merged**